lib/config/configuration.go

```diff
@@ -1012,6 +1012,13 @@ func Configure(clf *CommandLineFlags, cfg *service.Config) error {
 		}
 	}
 
+	// If this process is trying to join a cluster as an application service,
+	// make sure application name and URI are provided.
+	if utils.SliceContainsStr(splitRoles(clf.Roles), defaults.RoleApp) &&
+		(clf.AppName == "" || clf.AppURI == "") {
+		return trace.BadParameter("application name and URI flags are required to join cluster")
```

Comment:
To make the message more helpful to the user, I would specify which flags they need to provide e.g.

```
application name (--app-name) and URI (--app-uri) flags are both required to join application proxy to the cluster
```